nucypher/blockchain/eth/agents.py

```diff
@@ -356,69 +336,77 @@ def get_global_locked_tokens(self, at_period: int = None) -> int:
         """
         Returns the amount of tokens this staker has locked
         for a given duration in periods measured from the current period forwards.
         """
         if periods < 0:
             raise ValueError(f"Periods value must not be negative, Got '{periods}'.")
-        return self.contract.functions.getLockedTokens(staker_address, periods).call()
+        return NuNits(self.contract.functions.getLockedTokens(staker_address, periods).call())
 
     @contract_api(CONTRACT_CALL)
-    def owned_tokens(self, staker_address: str) -> int:
+    def owned_tokens(self, staker_address: ChecksumAddress) -> NuNits:
         """
         Returns all tokens that belong to staker_address, including locked, unlocked and rewards.
         """
-        return self.contract.functions.getAllTokens(staker_address).call()
+        return NuNits(self.contract.functions.getAllTokens(staker_address).call())
 
     @contract_api(CONTRACT_CALL)
-    def get_substake_info(self, staker_address: str, stake_index: int) -> Tuple[int, int, int]:
+    def get_substake_info(self, staker_address: ChecksumAddress, stake_index: int) -> SubStakeInfo:
         first_period, *others, locked_value = self.contract.functions.getSubStakeInfo(staker_address, stake_index).call()
-        last_period = self.contract.functions.getLastPeriodOfSubStake(staker_address, stake_index).call()
-        return first_period, last_period, locked_value
+        last_period: Period = self.contract.functions.getLastPeriodOfSubStake(staker_address, stake_index).call()
+        return SubStakeInfo(first_period, last_period, locked_value)
 
     @contract_api(CONTRACT_CALL)
-    def get_raw_substake_info(self, staker_address: str, stake_index: int) -> Tuple[int, int, int, int]:
-        result = self.contract.functions.getSubStakeInfo(staker_address, stake_index).call()
+    def get_raw_substake_info(self, staker_address: ChecksumAddress, stake_index: int) -> RawSubStakeInfo:
+        result: RawSubStakeInfo = self.contract.functions.getSubStakeInfo(staker_address, stake_index).call()
         first_period, last_period, periods, locked = result
-        return first_period, last_period, periods, locked
+        return RawSubStakeInfo(first_period, last_period, periods, locked)
 
     @contract_api(CONTRACT_CALL)
-    def get_all_stakes(self, staker_address: str) -> Iterable[Tuple[int, int, int, int]]:
-        stakes_length = self.contract.functions.getSubStakesLength(staker_address).call()
+    def get_all_stakes(self, staker_address: ChecksumAddress) -> Iterable[RawSubStakeInfo]:
+        stakes_length: int = self.contract.functions.getSubStakesLength(staker_address).call()
         if stakes_length == 0:
             return iter(())  # Empty iterable, There are no stakes
         for stake_index in range(stakes_length):
             yield self.get_substake_info(staker_address=staker_address, stake_index=stake_index)
```

Comment:
`get_substake_info` returns `SubStakeInfo` so returning type of this function will be `Iterable[SubStakeInfo]`